src/audio/base_fw.c

```diff
@@ -256,6 +265,19 @@ static uint32_t basefw_set_system_time(uint32_t param_id,
 	global_system_time_info.dsp_time.val_l = (uint32_t)(current_dsp_time);
 	global_system_time_info.dsp_time.val_u = (uint32_t)(current_dsp_time >> 32);
 
+	/* use default timestamp if 64bit is not enabled since 64bit is necessary for host time */
+	if (!IS_ENABLED(CONFIG_LOG_TIMESTAMP_64BIT)) {
+		LOG_WRN("64bits timestamp is disabled, so use default timestamp");
+		return IPC4_SUCCESS;
+	}
+
+	host_time = global_system_time_info.host_time.val_l |
+			((uint64_t)global_system_time_info.host_time.val_u << 32);
+	host_time_cycle = k_us_to_cyc_ceil64(host_time);
+	time_delta = host_time_cycle - current_dsp_cycle;
```

Comment:
The naming is quite confusing:
current_dsp_cycle - cycles
current_dsp_time - time in us
host_time - time in us
host_time_cycle - cycles
time_delta - cycles
?

why not:
current_dsp_cycle - cycles
current_dsp_time - time in us
host_time - time in us
host_cycle - cycles
cycle_delta - cycles

Reply:
unit seems not aligned.

Reply:
updated, thanks